noxfile.py

```diff
@@ -366,12 +363,46 @@ def build_dists(session: nox.Session) -> List[str]:
             "Remove them and try again",
         )
 
+    session.log("# Install build environment")
+    build_venv = Path(session.create_tmp()) / "build-venv"
+    session.run("python", "-m", "venv", "--without-pip", build_venv, silent=True)
+    build_python = build_venv / "bin" / "python"
+    session.run(
+        "python",
+        "-m",
+        "pip",
+        "--python",
+        build_python,
+        "install",
+        "-r",
+        HERE / "build-requirements.txt",
+        "--no-deps",
+        "--only-binary=:all:",
+        "--require-hashes",
+        silent=True,
+    )
+
     session.log("# Build distributions")
-    session.run("python", "-m", "build", silent=True)
+    session.run(
+        build_python,
+        "-m",
+        "build",
+        "--no-isolation",
+        silent=True,
+        external=True,
+    )
     produced_dists = glob.glob("dist/*")
 
     session.log(f"# Verify distributions: {', '.join(produced_dists)}")
-    session.run("twine", "check", *produced_dists, silent=True)
+    session.run(
+        build_python,
+        "-m",
+        "twine",
+        "check",
```

Comment:
(not really related to the refactoring in this PR, but it should always be strict — would be good to put into a separate PR, I suppose)
```suggestion
        "check",
        "--strict",
```